File: exercism/ocaml/difference_of_squares.ml
let square_of_sum n =
  let m = n * (n + 1) / 2 in
  m * m

let sum_of_squares n = n * (n + 1) * ((2 * n) + 1) / 6

let difference_of_squares n = square_of_sum n - sum_of_squares n

File: exercism/c/difference_of_squares.c
#include "difference_of_squares.h"
#include<math.h>

unsigned int square_of_sum(unsigned int number) {
  unsigned int m = (number * (number + 1) / 2);
  return m*m;
}

unsigned int sum_of_squares(unsigned int number) {
  return (number * (number + 1) * (2 * number + 1) / 6);
}

unsigned int difference_of_squares(unsigned int number) {
  return square_of_sum(number) - sum_of_squares(number);
}
